File: MAP(Advanced Programming Methods)/countdownlatch/src/utils/ExeStack.java
package utils;

import java.util.ArrayDeque;
import java.util.Deque;

public class ExeStack<V> implements IExeStack<V> {

    private Deque<V> stack;

    public ExeStack()
    {
        this.stack = new ArrayDeque<>();
    }

    @Override
    public void push(V el) {
        this.stack.push(el);
    }

    @Override
    public V pop() {
        return this.stack.pop();
    }

    @Override
    public boolean isEmpty() {
        return this.stack.isEmpty();
    }

    @Override
    public V top() {
        return this.stack.peek();
    }

    @Override
    public Iterable<V> getAll() {
        return this.stack;
    }

    @Override
    public String toString() {
        StringBuilder string = new StringBuilder();
        string.append("ExeStack = {");

        if (!stack.isEmpty()) {
            string.append("\n");
        }

        for (V statement : stack) {
            string.append("   " + statement.toString() + "\n");
        }

        string.append("}");
        return string.toString();
    }
}

File: MAP(Advanced Programming Methods)/LabAssig7/LabAssig7/Model/ADT/ExeStack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Collections;
namespace LabAssig7.Model.ADT
{
    public class ExeStack<V>:IExeStack<V>
    {
        private Stack<V> stack;

        public ExeStack()
        {
            this.stack = new Stack<V>();
        }

        public void Push(V el)
        {
            this.stack.Push(el);
        }
        public V pop()
        {
            return this.stack.Pop();
        }
        public bool IsEmpty()
        {
            return this.stack.Count == 0;
        }
        public IEnumerator GetEnumerator()
        {
            return stack.GetEnumerator();
        }
        public override string ToString()
        {
            StringBuilder buff = new StringBuilder();

            foreach(V el in this.stack)
            {
                buff.Append(el);
                buff.Append(", ");
            }
            buff.Append("\n");
            return buff.ToString();
        }
    }
}
